summary_workflows/quantification/match_with_gt.py

```diff
@@ -0,0 +1,133 @@
+
+    # memory file-handle to pass output to pandas without writing to disk
+    out_handle = StringIO(out.stdout.decode())
+    out = pd.read_csv(out_handle, delimiter='\t', header=None, dtype={0: str})
+    
+    return(out)
+
+def overlap(coords1, coords2):
+    """
+    Calculates number of points overlapping when the input are two intervals as tuples.
+    Input must be coords1=(int, int),  coords2=(int, int).
+    """
+    return(len(np.intersect1d(np.arange(coords1[0], coords1[1]), np.arange(coords2[0], coords2[1]))))
+
+
+print('-------------------------------------------------')
+
+
+# bedtools window with specified parameter
+out = bedtools_window(f_PD, f_GT, window)
+# label columns
+out.rename({0: 'chrom_p', 1: 'chromStart_p', 2: 'chromEnd_p', 3: 'name_p', 4: 'score_p', 5: 'strand_p', 6: 'chrom_g', 7: 'chromStart_g', 8: 'chromEnd_g', 9: 'name_g', 10: 'score_g', 11: 'strand_g'}, axis=1, inplace=True)
+
+
+# initialize weight column with default 1 for a perfect match
+out['weight'] = [1]*len(out)
+
+# counter for logging
+weights_counter = 0
+
+# find rows in sample that are not unique 
+# -> extended predicted site overlaps multiple ground truth sites
+not_unq_mask = out.duplicated(['chrom_p', 'chromStart_p', 'chromEnd_p', 'strand_p'], keep=False)
+
+if np.sum(not_unq_mask) > 0: # otherwise there was no overlap
+        
+    not_unq = out[not_unq_mask]
+    
+    # remove non-unique sites
+    out.drop(out.index[not_unq_mask], inplace=True, axis=0)
+        
+    # get the names of non-unique columns
+    unq = np.unique([(i, j, k, l) for i, j, k, l in zip(not_unq['chrom_p'], not_unq['chromStart_p'], not_unq['chromEnd_p'], not_unq['strand_p'])], axis=0)
+
+    for i in unq:
+        # split expression between ground truth sites, according to vicinity
+
+        # re-name sites that have been split
+
+
+    
+        pass
+
+
+
+        # non_unq_context = not_unq[(not_unq['chrom_p']==i[0]) & (not_unq['chromStart_p']==int(i[1])) & (not_unq['chromEnd_p']==int(i[2])) & (not_unq['strand_p']==i[3])].copy()
+        
+        # olps = []
+        # for idx, gt_site in non_unq_context.iterrows():
+        #     frac_overlap = overlap((gt_site['chromStart_p']-window, gt_site['chromEnd_p']+window), (gt_site['chromStart_g']-window, gt_site['chromEnd_g']+window))/(gt_site['chromEnd_p']-gt_site['chromStart_p']+2*window)
+        #     olps.append(frac_overlap)
+        #     weights_counter += 1
+        
+        # weights = np.array(olps)/np.sum(olps)
+        # non_unq_context['weight'] = weights
+        
+        # # add back the removed sites including weights
+        # out = out.append(non_unq_context)
+
+# find sites with no overlap given the window
+out_rev = bedtools_window(f_PD, f_GT, window, reverse=True)
+out_rev.rename({0: 'chrom_p', 1: 'chromStart_p', 2: 'chromEnd_p', 3: 'name_p', 4: 'score_p', 5: 'strand_p', 6: 'chrom_g', 7: 'chromStart_g', 8: 'chromEnd_g', 9: 'name_g', 10: 'score_g', 11: 'strand_g'}, axis=1, inplace=True)
```

Comment:
You could do the renaming in `bedtools_window` directly, then you don't have to write the same code twice.